### PR TITLE
handle ocaml 4.12. + improve error message

### PR DESCRIPTION
**Pull request by zoggy**
and improve printing of fatal errors.

### CODE REVIEW

File: src/parse_structure.ml [hunk 1]
```diff
@@ -52,7 +52,12 @@ let type_with_init lst =
   try
     ret @@
       extract_typedtree @@
-        Typemod.type_structure (init_env ()) lst Location.none
+        Typemod.type_structure (init_env ()) lst
+#if OCAML_VERSION >= (4, 12, 0)
```
Comment (nobrakal) [inline src/parse_structure.ml]: Is it possible to use `#if OCAML_VERSION < (4, 12, 0)` to avoid the empty branch ?

File: src/parse_structure.ml [hunk 2]
```diff
@@ -63,11 +68,15 @@ let simplify_lambda lambda =
 #endif
 
 let transl_exp expr =
+#if OCAML_VERSION >= (4, 12, 0)
+  Translcore.transl_exp ~scopes:Debuginfo.Scoped_location.empty_scopes expr
+#else
 #if OCAML_VERSION >= (4, 11, 0)
```
Comment (nobrakal) [inline src/parse_structure.ml]: I think you can use the `#elif` statement here to avoid the two `#endif` at the end.

**Review by nobrakal — COMMENTED** [review]
Thanks for the PR! I left two little comments :)

File: src/parse_structure.ml (hunk 1 above)
Comment (zoggy) [reply src/parse_structure.ml]: Sure !

File: src/parse_structure.ml (hunk 2 above)
Comment (zoggy) [reply src/parse_structure.ml]: Indeed :-)

**Comment by nobrakal** [comment]
Many thanks, I am merging now :)

**Pull request merged**